src/Microsoft.Framework.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -31,23 +31,32 @@ public static void ExecuteWithFileLocked(string filePath, Action<bool> action)
 
         public async static Task<T> ExecuteWithFileLocked<T>(string filePath, Func<bool, Task<T>> action)
         {
-            var createdNew = false;
-            var fileLock = new Semaphore(initialCount: 0, maximumCount: 1, name: FilePathToLockName(filePath),
-                createdNew: out createdNew);
-            try
+            for (var i = 0; i < 3; ++i)
```

Comment:
`3 * 5` seconds seconds doesn't seem like a very long time to wait for a package to extract. Have you tried extracting a large sized package?